server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java

```diff
@@ -211,7 +218,10 @@ public Invocations getInvocations() {
       if (template != null && !template.isEmpty()) {
          String templateName = template.iterator().next();
          return CompletableFuture.supplyAsync(() -> {
-            administration.createCache(cacheName, templateName);
+            Security.doAs(request.getSubject(), (PrivilegedAction<Void>) () -> {
```

Comment:
This should be unnecessary: the administration object should already have the Subject set on it.